frontend/src/components/accountPage/SSHKeysForm/SSHKeysForm.tsx

```diff
@@ -0,0 +1,128 @@
+import { Input, Form, Button, Row } from 'antd';
+import { FC, useState } from 'react';
+
+export interface ISSHKeysFormProps {
+  onSaveKey: (newKey: { name: string; key: string }) => void;
+  onCancel: () => void;
+}
+
+const acceptedAlgorithms = [
+  'sk-ecdsa-sha2-nistp256@openssh.com',
+  'ecdsa-sha2-nistp256',
+  'ecdsa-sha2-nistp384',
+  'ecdsa-sha2-nistp521',
+  'sk-ssh-ed25519@openssh.com',
+  'ssh-ed25519',
+  'ssh-dss',
+  'ssh-rsa',
+];
+
+const SSHKeysForm: FC<ISSHKeysFormProps> = props => {
+  const { onSaveKey, onCancel } = props;
+  const [validForm, setValidForm] = useState<boolean>(false);
+  const [form] = Form.useForm();
+
+  const cancelForm = () => {
+    form.resetFields();
+    onCancel();
+  };
+  const validateSSHKey = async (rules: any, key: string) => {
+    setValidForm(false);
+    if (!key) throw new Error('Key field is mandatory');
+    const result = key.split(/\s+/); // split regardless the amount of spaces
+    if (result.length !== 2 && result.length !== 3)
+      throw new Error('Invalid key format');
+    if (result[1] === '') throw new Error('Empty SSH key');
+    if (!acceptedAlgorithms.includes(result[0]))
+      throw new Error('Unrecognized SSH key format');
+
+    setValidForm(true);
+  };
+
+  // Used to disable Save button
+  const handleChange = async () => {
+    try {
+      if (
+        form.getFieldValue('key') !== undefined &&
+        form.getFieldValue('name') !== undefined
+      ) {
+        await form.validateFields(['key', 'name']);
+        setValidForm(true);
+      } else if (form.getFieldValue('key') !== undefined)
+        await form.validateFields(['key']);
+      else await form.validateFields(['name']);
+    } catch (_) {
+      setValidForm(false);
+    }
+  };
+  const submitForm = ({ name, key }: { name: string; key: string }) => {
+    // the name field is appended to the key
+    const comment = key.split(/\s+/)[2];
+    name = name.trim();
+    if (comment) key = `${key}:${name}`;
+    else key = `${key} ${name}`;
+
+    onSaveKey({ name, key });
+    form.resetFields();
+  };
+
+  return (
+    <Form
+      form={form}
+      labelCol={{ span: 4 }}
+      wrapperCol={{ span: 24 }}
+      onFinish={submitForm}
+      onChange={handleChange}
```

Comment:
I don't remember if there was any issue with using `onValuesChange`, it seems from documentation that it's not really advisable to use onChange... Also `onValuesChange` could make code much cleaner...